service/src/main/java/bio/terra/workspace/service/resource/controlled/model/ControlledResourceFields.java

```diff
@@ -0,0 +1,233 @@
+package bio.terra.workspace.service.resource.controlled.model;
+
+import bio.terra.workspace.db.model.DbResource;
+import bio.terra.workspace.service.iam.model.ControlledResourceIamRole;
+import bio.terra.workspace.service.resource.ValidationUtils;
+import bio.terra.workspace.service.resource.model.CloningInstructions;
+import java.util.Optional;
+import java.util.UUID;
+import javax.annotation.Nullable;
+
+/**
+ * ControlledResourceFields is used as a way to collect common resources for a controlled resource.
+ * That way, we can more easily add common resource parameters to all resources without visiting
+ * each implementation. Although for safe serialization, we still have to visit each @JsonCreator
+ * resource constructor and add the parameters.
+ *
+ * <p>This allows us to make controller code that processes the common parts of the API input and
+ * uses the methods in this class to populate the builder.
+ */
+public class ControlledResourceFields {
+  private final UUID workspaceId;
+  private final UUID resourceId;
+  private final String name;
+  @Nullable private final String description;
+  private final CloningInstructions cloningInstructions;
+  @Nullable private final String assignedUser;
+  // We hold the iamRole to simplify the controller flow. It is not retained in the
+  // controlled object.
+  @Nullable private final ControlledResourceIamRole iamRole;
+  // Default value is NOT_APPLICABLE for shared resources and INITIALIZING for private resources.
+  @Nullable private final PrivateResourceState privateResourceState;
+  private final AccessScopeType accessScope;
+  private final ManagedByType managedBy;
+  @Nullable private final UUID applicationId;
+
+  /** construct from database resource */
+  public ControlledResourceFields(DbResource dbResource) {
+    workspaceId = dbResource.getWorkspaceId();
+    resourceId = dbResource.getResourceId();
+    name = dbResource.getName().orElse(null);
```

Comment:
Good catch. That has been wrong in DbResource for a long time.

Reply:
For DbResource, accessScope and managedBy are Nullable because the class is used to describe referenced resources also. However, if they are null in this case, there should be an error.

I think I'll add getRequired___ getters for those and throw.